homeassistant/components/velbus/config_flow.py

```diff
@@ -46,24 +48,76 @@ async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> ConfigFlowResult:
         """Step when user initializes a integration."""
-        self._errors = {}
+        return self.async_show_menu(
+            step_id="user", menu_options=["network", "usbselect"]
+        )
+
+    async def async_step_network(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle network step."""
+        if user_input is not None:
+            self._title = "Velbus Network"
+            if user_input[CONF_TLS] == "yes":
+                self._device = "tls://"
+            else:
+                self._device = ""
+            if user_input[CONF_PASSWORD] != "":
+                self._device += f"{user_input[CONF_PASSWORD]}@"
+            self._device += f"{user_input[CONF_HOST]}:{user_input[CONF_PORT]}"
+            self._async_abort_entries_match({CONF_PORT: self._device})
+            if await self._test_connection():
+                return self._create_device()
+        else:
+            user_input = {}
+            user_input[CONF_TLS] = "yes"
+            user_input[CONF_HOST] = ""
+            user_input[CONF_PORT] = 27015
+            user_input[CONF_PASSWORD] = ""
+
+        return self.async_show_form(
+            step_id="network",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_TLS, default=user_input[CONF_TLS]): vol.In(
+                        ["yes", "no"]
+                    ),
+                    vol.Required(CONF_HOST, default=user_input[CONF_HOST]): str,
+                    vol.Required(CONF_PORT, default=user_input[CONF_PORT]): int,
+                    vol.Optional(CONF_PASSWORD, default=user_input[CONF_PASSWORD]): str,
```

Comment:
Please use appropriate selectors instead.
Use `self.add_suggested_values_to_schema(` to add suggested values